cns/hnsclient/hnsclient_windows.go

```diff
@@ -600,6 +603,15 @@ func CreateHostNCApipaEndpoint(
 	return endpoint.Id, nil
 }
 
+// adhocAdjustIPConfig applies adhoc change on gw IP address
+func adhocAdjustIPConfig(localIPConfiguration *cns.IPConfiguration) {
```

Comment:
Instead of adding this, is it possible to change the localIPConfiguration we pass into CreateHostNCApipaEndpoint? Is this a temporary solution? I feel like it might be a bit confusing if we pass in a local ip configuration and then the code changes what we directed cns to do to something else.

Reply:
the caller of CreateHostNCApipaEndpoint is https://github.com/Azure/azure-container-networking/blob/3ed0bcd0cae71c44396e518e887c5df3d8e81f75/cns/restserver/api.go#L1151. where is the place we handle the request. But I dont think that is the place to have this kind of logic to update the gw. what do you think? @QxBytes

Reply:
Ideally I would like to get a different value from networkContainerDetails but I see that in order to do that we'd need to store the  .2 in the cns state, which leads us back to the long term fix of needing to get the gw address from elsewhere and cns consuming it.